sal: sid_pal: store keys in KMU instead of trusted storage (#780)
nrfconnect/sdk-sidewalk: doc/releases_and_migration/migration_guide_addon_changelog.rst

.. note::
   Use ED25519 for new designs on nRF54L Series platforms because of its shorter verification time and smaller key size.

Persistent Sidewalk keys in KMU
*******************************

> This release stores persistent Sidewalk cryptographic keys in the Key Management Unit (KMU) by default on supported nRF54L Series platforms.

Reviewer: ```suggestion
Persistent Sidewalk cryptographic keys are now stored in the Key Management Unit (KMU) by default on supported nRF54L Series platforms.
```

> The feature is controlled by the ``CONFIG_SIDEWALK_CRYPTO_PSA_KEY_STORAGE_KMU`` Kconfig option.

Reviewer: ```suggestion
Use the ``CONFIG_SIDEWALK_CRYPTO_PSA_KEY_STORAGE_KMU`` Kconfig option to control this feature.
```

If you update devices that already have keys stored by an earlier firmware version, set ``CONFIG_SIDEWALK_CRYPTO_PSA_KEY_STORAGE_KMU=n`` to keep the keys accessible.

> Switching an already-provisioned product from the default PSA trusted storage backend to KMU changes the PSA key IDs used by Sidewalk, as well as the storage location, so existing keys will not be found.

Reviewer: ```suggestion
Switching a provisioned product from the PSA trusted storage to KMU changes both the storage location and the PSA key IDs, so Sidewalk cannot find the existing keys.
```

> For new products, use the KMU default and make sure that your application does not use the same KMU slots for other keys.

Reviewer: ```suggestion
For new products, use the KMU default. 
Make sure that your application does not use the same KMU slots for other keys.
```

> Sidewalk starts at ``CONFIG_SIDEWALK_CRYPTO_PSA_KEY_STORAGE_KMU_SLOT_START`` and reserves seven consecutive KMU slots:

Reviewer: ```suggestion
Sidewalk starts at the slot defined by the ``CONFIG_SIDEWALK_CRYPTO_PSA_KEY_STORAGE_KMU_SLOT_START`` Kconfig option and reserves seven consecutive KMU slots:
```

* Two slots for the manufacturing ED25519 private key.
* Two slots for the manufacturing secp256r1 private key.
* One slot each for the WAN master, app key, and device-to-device AES keys.

> 

Reviewer 2: This is migration guide section, so maybe it is not the best place.. And maybe out-of-scope of this PR: but @ktaborowski  @Damian-Nordic do we have a documentation about what kind of keys we do store and how they are provisioned - e.g. from mfg partition or during Sidewalk registration? 

I would love to see such a short section somewhere. Actually this section already covers it in some degree - but it is in migration guide, which won't be found later.
